daemon/daemon.go

```diff
@@ -923,6 +1001,7 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
 	d.RegistryService = registryService
 	d.EventsService = eventsService
 	d.volumeDriver = volumesDriver
+	d.root = config.Root
```

Comment:
ugh... right now the only path available in daemon is `daemon.repository`... use this for the volume config migrations

we could use `daemon.repository`,but that assumes repository is at the daemon root... probably not the worst thing, but probably slightly worse than adding this.